src/chai/ManagedArray.inl

```diff
@@ -333,29 +385,20 @@ CHAI_INLINE
 CHAI_HOST
 void ManagedArray<T>::move(ExecutionSpace space)
 {
-  ExecutionSpace prev_space = m_pointer_record->m_last_space;
-
-  /* When moving from CPU to GPU we need to move the inner arrays before the outer array. */
-  if (prev_space == CPU) {
-    moveInnerImpl(space);
+  if (m_resource_manager == nullptr) {
```

Comment:
If m_resource_manager is set in every host constructor, then we don't need to check in all these host methods.